### PR TITLE
ci: run ci-release workflow on release branches

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2364

Note this needs to be backported to v1.x before it will take effect.

## Testing

Tested on my fork. Before this PR, only docker-security-publish workflows run on commits after they merge to the v1.x branch. After this PR, the ci-release workflows also run.

Before | After
--- | ---
Screenshot 2023-08-25 at 4 00 00 PM | Screenshot 2023-08-25 at 4 00 06 PM

### CODE REVIEW

File: .github/workflows/ci-release.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ on:
   push:
     branches:
       - main
+      - v1.x
     tags:
       - "v[0-9]+.[0-9]+.[0-9]+"
```
Comment (evan-forbes) [inline .github/workflows/ci-release.yml:11]: could we generalize this branch to be any versioned branch like the tags below?

**Review by evan-forbes — COMMENTED** [review]
:+1:

File: .github/workflows/ci-release.yml (hunk 1 above)
Comment (evan-forbes) [reply .github/workflows/ci-release.yml:11]: defo not blocking

**Review by cmwaters — APPROVED** [review]

File: .github/workflows/ci-release.yml [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 
-# Trigger on push events to main (i.e. merges), pushing new semantic version
-# tags, all PRs, and manual triggers
+# Run this workflow on push events (i.e. PR merge) to main or release branches,
+# push events for new semantic version tags, all PRs, and manual triggers.
 on:
   push:
     branches:
       - main
+      - "v[0-9]+.x"
```
Comment (evan-forbes) [inline .github/workflows/ci-release.yml:9]: maybe also
```suggestion
      - "v[0-9]+.x"
      - "v[0-9]+.[0-9]+.x"
```
stupid formatting question, does 0-9 just mean any number, or literally 0 through 9?

**Review by evan-forbes — APPROVED** [review]

File: .github/workflows/ci-release.yml (hunk 2 above)
Comment (rootulp) [reply .github/workflows/ci-release.yml:9]: - `[0-9]` means any digit
- `+` means 1 or more

hmm I think we only need `"v[0-9]+.[0-9]+.x"` if we intend on maintaining release branches for minor versions. In other words:

- `v1.x`
- `v1.0.x`
- `v1.1.x`
- `v1.2.x`

but I thought we were only planning on maintaining release branches per major version

**Pull request merged**